java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageSerializer.java

```diff
@@ -266,8 +266,15 @@ public static ArrowBlock serialize(WriteChannel out, ArrowRecordBatch batch, Ipc
     long bufferLength = writeBatchBuffers(out, batch);
     Preconditions.checkArgument(bufferLength % 8 == 0, "out is not aligned");
 
+    long compLength = 0L;
+    if (batch.getBodyCompression() != null) {
+      compLength = writeCompressionBody(out, batch.getBodyCompression());
+      Preconditions.checkArgument(compLength == ArrowBodyCompression.BODY_COMPRESSION_LENGTH,
+          "deserialized compression body length not equal to ArrowBodyCompression#BODY_COMPRESSION_LENGTH");
```

Comment:
included sizes here would be useful.

Reply:
This check has been removed too. Thanks.